.github/workflows/lint_and_test.yaml

```diff
@@ -0,0 +1,35 @@
+name: Python package
+
+on: [push]
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        python-version: ["3.9", "3.10", "3.11"]
+
+    steps:
+      - uses: actions/checkout@v3
+      - name: Set up Python ${{ matrix.python-version }}
+        uses: actions/setup-python@v4
+        with:
+          python-version: ${{ matrix.python-version }}
+          cache: "pip"
+      - name: Install dependencies
+        run: |
+          python -m pip install --upgrade pip
+          pip install ruff pytest mypy
+          if [ -f test-requirements.txt ]; then pip install -r test-requirements.txt; fi
```

Comment:
I think we should add `ruff`, `pytest` and `mypy` to test-requirements.txt

Reply:
I was going to say that "those are linting requirements more than test requirements", but since we do both at once, that probably does make sense